builtin/credential/aws/path_role.go

```diff
@@ -20,6 +20,12 @@ func pathRole(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "Name of the role.",
 			},
+			"allowed_auth_types": {
```

Comment:
This is a good question and it's not a super straightforward answer. In the past the philosophy is "a new Vault installation should work according to current best practices", so what we've done is used upgrade paths to not modify behavior of existing roles but have new roles gain new behavior (or roles that are then updated), and then we document this change in the upgrade guide.

I think the IAM functionality will be super useful, and I think having it be available by default is the nicest option from a user perspective. We do publish upgrade guides for a reason. So my gut says we should add an upgrade path (which I think would be, type is ec2 and inference is off since if iam isn't enabled binds are always checked?) and default it to `ec2,iam` (with inference checking on?)

I'll ask the team too.